lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -445,61 +406,139 @@ function Get-CharmServices {
             )
         }
     }
-
+    $s2dContextGen = @{
+        "generator" = (Get-Item "function:Get-S2DCSVContext").ScriptBlock
+        "relation" = "csv"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'csv')) {
+        $s2dContextGen['mandatory'] = $true
+    }
+    $freeRdpContextGen = @{
+        "generator" = (Get-Item "function:Get-FreeRDPContext").ScriptBlock
+        "relation" = "free-rdp"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'free-rdp')) {
+        $freeRdpContextGen['mandatory'] = $true
```

Comment:
not a fan of this. It's either mandatory or not. If a relation is created for a non-mandatory interface, we should just report if it's incomplete or not. The mandatory flag should be ```readonly```.